### DISCUSSION TITLE
"Go to Definition" doesnt work for private methods of stdlib

### DISCUSSION THREAD

**Discussion by timhughes** [microsoft/pylance-release#4765] [General]
"Go to Definition" doesn't work for private methods of stdlib. In the following code example "Go to definition" doesn't work for `self._last_mtime`. It is probably related to **pyright** only looking at **typeshed** for symbols in stdlib https://github.com/microsoft/pyright/issues/5825

The "Go to definition" works correctly if I change `"python.languageServer": "Pylance"` server to `"Jedi"`

I don't know if this is a bug or improvement for PyLance or a config option I can change somewhere

```python
from gzip import _GzipReader


class _MyGzipReader(_GzipReader):
    def __init__(self, fp):
        super().__init__(fp)
        self._my_last_mtime = self._last_mtime
```
![Screenshot from 2023-08-27 02-02-28](https://github.com/microsoft/pylance-release/assets/1203343/a78b168c-fe73-483d-b2f4-d52cc6043ebf)

**Comment by erictraut** [comment]
This is functionality that pylance adds on top of pyright. It requires some sophisticated heuristics to find associations between typeshed stubs and implementation files. We have no plans to implement this in pyright. If you want this functionality, you can use pylance.

**Reply by timhughes** [reply]
I am using PyLance but it doesn't work. The functionality only works if I switch to Jedi. I am talking about the part where you hold down CTRL and select the member and it goes to the definition in the .py file

**Reply by erictraut** [reply]
Oh, that's because this symbol is defined as "not present" in the type stub. The type stub is considered the source of truth for type information, and pylance won't look for it in the source file if the type stub says that it's not present.

**Reply by timhughes** [reply]
Does that mean this is a "Wontfix" or a feature request? I just want to be able to click through to the implementation

**Reply by erictraut** [reply]
You could file a bug report / PR in the typeshed project if you think that this symbol should be included in the stub file. Or you could file a feature request in the [pylance-release project](https://github.com/microsoft/pylance-release) and see if the pylance team is willing to add extra logic to ignore the type information in the type stub and look for private symbols in the implementation. It's a "won't fix" for pyright, since pyright is focused on type checking and this isn't a type checking feature.
